web/sharings/sharings.go

```diff
@@ -494,3 +538,18 @@ func wrapErrors(err error) error {
 	}
 	return err
 }
+
+func doesDoctypeExist(ins *instance.Instance, doctype string) (bool, error) {
+	doctypes, err := couchdb.AllDoctypes(ins)
```

Comment:
I think it would be better to use `couchdb.DbStatus` to check if a doctype exists or not, instead of `AllDoctypes` (which is not very efficient).